File: Programming Basics with Python/06 Nested loops/More exercises/03_Lucky_Numbers.py
number = int(input())
for first_digit in range(1, 10):
    for second_digit in range(1, 10):
        for third_digit in range(1, 10):
            for fourth_digit in range(1, 10):
                if number % (first_digit + second_digit) == 0 and (first_digit + second_digit) == (
                        third_digit + fourth_digit):
                    print(f"{first_digit}{second_digit}{third_digit}{fourth_digit}", end=" ")

File: JS Basics/06 Nested Loops - More Exercises/03. Lucky Numbers.js
function lucky(input) {
    let luckyNumber = Number(input[0]);
    let print = "";
    for (let firstDigit = 1; firstDigit <= 9; firstDigit++) {
        for (let secondDigit = 1; secondDigit <= 9; secondDigit++) {
            for (let thirdtDigit = 1; thirdtDigit <= 9; thirdtDigit++) {
                for (let fourthDigit = 1; fourthDigit <= 9; fourthDigit++) {
                    if (((firstDigit + secondDigit) == (thirdtDigit + fourthDigit)) && (luckyNumber % (firstDigit + secondDigit) == 0)) {
                        print += (`${firstDigit}${secondDigit}${thirdtDigit}${fourthDigit} `);

                    }
                }
            }
        }
    }
    console.log(print)
}
lucky([7])